### PR TITLE
Terraform GitHub Actions updates

### PR DESCRIPTION
**Pull request by sudomateo**
Updated to reflect YAML syntax
Reorganized content layout

### CODE REVIEW

**Comment by sudomateo** [comment]
I meant to make this a draft pull request. I still have a few updates I would like to make.

**Comment by sudomateo** [comment]
This is ready for review.

File: content/source/docs/github-actions/configuration/index.html.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+## Outputs
+
+Outputs are used to pass information to subsequent GitHub Actions steps.
+
+* `tf_actions_plan_has_changes` - Whether or not the Terraform plan contained changes.
+
+## Secrets
+
+Secrets are similar to inputs except that they are encrypted and only used by GitHub Actions. It's a convenient way to keep sensitive data out of the GitHub Actions workflow YAML file.
```
[the remark this answers is not in the document]
Comment (sudomateo) [reply content/source/docs/github-actions/configuration/index.html.md]: I updated this with your suggestion, rewording it in a few places.

File: content/source/docs/github-actions/configuration/index.html.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+Inputs configure Terraform GitHub Actions to perform different actions.
+
+* `tf_actions_version` - (Required) The Terraform version to install and execute.
+* `tf_actions_subcommand` - (Required) The Terraform subcommand to execute. Valid values are `fmt`, `init`, `validate`, `plan`, and `apply`.
+* `tf_actions_working_dir` - (Optional) The working directory to change into before executing Terraform subcommands. Defaults to `.` which means use the root of the GitHub repository.
+* `tf_actions_comment` - (Optional) Whether or not to comment on GitHub pull requests. Defaults to `true`.
+
+## Outputs
+
+Outputs are used to pass information to subsequent GitHub Actions steps.
```
[the remark this answers is not in the document]
Comment (sudomateo) [reply content/source/docs/github-actions/configuration/index.html.md]: Updated this with a brief blurb about outputs and then a link to the official GitHub Actions docs.

**Comment by sudomateo** [comment]
Updated with changes based on your feedback. I went with the name "Common Actions" instead of "Workflows". I'm open to changing it if needed.

I also updated `content/redirects.txt` to redirect all of the changed URLs to their new location.

**Comment by sudomateo** [comment]
The TravisCI build is complaining about missing links.
```
http://127.0.0.1:4567/docs/providers/azurerm/auth/azure_cli.html:
Remote file does not exist -- broken link!!!
http://127.0.0.1:4567/docs/providers/azurerm/auth/managed_service_identity.html:
Remote file does not exist -- broken link!!!
http://127.0.0.1:4567/docs/providers/azurerm/auth/service_principal_client_secret.html:
Remote file does not exist -- broken link!!!
```

Seems like these links are now located at the following places respectively.
```
/docs/providers/azurerm/guides/azure_cli.html
/docs/providers/azurerm/guides/managed_service_identity.html
/docs/providers/azurerm/auth/service_principal_client_secret.html
```

**Comment by nfagerlund** [comment]
Looking into those azure things right now, I think I found the issue.

File: content/source/docs/github-actions/common-actions/arguments.html.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
+---
+layout: "github-actions"
+page_title: "Arguments- Terraform GitHub Actions"
```
Comment (nfagerlund) [inline content/source/docs/github-actions/common-actions/arguments.html.md]:
```suggestion
page_title: "Arguments - Terraform GitHub Actions"
```

File: content/source/docs/github-actions/common-actions/working-directory.html.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
+---
+layout: "github-actions"
+page_title: "Working Directory - Terraform GitHub Actions"
+sidebar_current: "docs-github-actions-common-actions-working-directory"
```
Comment (nfagerlund) [inline content/source/docs/github-actions/common-actions/working-directory.html.md]: Oh argh, I forgot to mention on the last pass of reviews that we don't need sidebar_current at all anymore. 😭 You can leave it in, but there's no need to keep updating it, it's inert now.

File: content/source/docs/github-actions/common-actions/index.html.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
+---
+layout: "github-actions"
+page_title: "Workflows - Terraform GitHub Actions"
+sidebar_current: "docs-github-actions-common-actions"
+---
+
+# Configuration
```
Comment (nfagerlund) [inline content/source/docs/github-actions/common-actions/index.html.md]:
```suggestion
# Common Tasks
```

File: content/source/docs/github-actions/common-actions/index.html.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
+---
+layout: "github-actions"
+page_title: "Workflows - Terraform GitHub Actions"
```
Comment (nfagerlund) [inline content/source/docs/github-actions/common-actions/index.html.md]:
```suggestion
page_title: "Common Tasks - Terraform GitHub Actions"
```

File: content/source/docs/github-actions/common-actions/index.html.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+sidebar_current: "docs-github-actions-common-actions"
+---
+
+# Configuration
+
+Terraform GitHub Actions are designed to be configurable for most workflows.
+
+See the links below for examples of common workflows.
```
Comment (nfagerlund) [inline content/source/docs/github-actions/common-actions/index.html.md]:
```suggestion
However, some Terraform features require slightly different configuration when used in GitHub Actions. See the links below for examples of how to use various Terraform features within GitHub Actions.
```

**Pull request merged**